torchvision/models/convnext.py

```diff
@@ -0,0 +1,201 @@
+from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Sequence
+
+import torch
+from torch import nn, Tensor
+from torch.nn import functional as F
+
+from .._internally_replaced_utils import load_state_dict_from_url
+from ..ops.misc import ConvNormActivation
+from ..ops.stochastic_depth import StochasticDepth
+from ..utils import _log_api_usage_once
+
+
+__all__ = [
+    "ConvNeXt",
+    "convnext_tiny",
+]
+
+
+model_urls: Dict[str, Optional[str]] = {}
+
+
+class LayerNorm2d(nn.LayerNorm):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        self.channels_last = kwargs.pop("channels_last", False)
+        super().__init__(*args, **kwargs)
+
+    def forward(self, x):
+        if not self.channels_last:
+            x = x.permute(0, 2, 3, 1)
+        x = F.layer_norm(x, self.normalized_shape, self.weight, self.bias, self.eps)
+        if not self.channels_last:
+            x = x.permute(0, 3, 1, 2)
+        return x
+
+
+class CNBlock(nn.Module):
+    def __init__(self, dim, layer_scale: float, stochastic_depth_prob: float, norm_layer: Callable[..., nn.Module]):
+        super().__init__()
+        self.block = nn.Sequential(
+            ConvNormActivation(
+                dim,
+                dim,
+                kernel_size=7,
+                groups=dim,
+                norm_layer=norm_layer,
+                activation_layer=None,
+                bias=True,  # TODO: check
```

Comment:
Can you talk about your decision using Bias on the first convolution of the ConvNeXt block? Have you experimented and found it necessary?

Reply:
Thanks for noting this. We have not tested (I'm going to launch a run with T model soon) but it should be totally unnecessary due to the LayerNorm after this.

Reply:
Thanks for confirming. Indeed in 99% of cases, the bias prior the Normalization layer is unnecessary. We've seen previously one [niche case in optical flow](https://github.com/pytorch/vision/blob/922db3086e654871c35cd80c2c01eabb65d78475/torchvision/models/optical_flow/raft.py#L35-L40) were a combination of transfer learning and freezing the BN made sense to have biases, so I just wanted to confirm there isn't anything like that here. Let's compare notes after you run the model, to confirm.

Reply:
The 99% of cases you have seen are BN, where the biases will become spurious parameters since they will be canceled out anyways. ConvNeXt uses LN, which means the Conv2D biases here will have an effect. I still believe it won't matter much for the final results but it is not equivalent. Only showing results with Tiny model is also not enough, we need to run all the model sizes to make sure the conclusion holds true when scaling up. My suggestion is to keep it True here.